Add GraalVM Reflection Configuration for MCPSchema

sdelamo opened the pull request:
This PR adds a test project for the MCP Server tools Weather demo in https://modelcontextprotocol.io/docs/develop/build-server

It adds reflect-config.json configuration for every Java record in https://github.com/modelcontextprotocol/java-sdk/blob/main/mcp/src/main/java/io/modelcontextprotocol/spec/McpSchema.java to make the native tests pass.

sdelamo committed:
Add GraalVM Reflection Configuration

This PR adds a test project for the MCP Server tools Weather demo in https://modelcontextprotocol.io/docs/develop/build-server

It adds reflect-config.json configuration for every Java record in https://github.com/modelcontextprotocol/java-sdk/blob/main/mcp/src/main/java/io/modelcontextprotocol/spec/McpSchema.java to make the native tests pass.

sdelamo committed:
add to appropiate folder

sdelamo committed:
remove java 21

sdelamo committed:
extract to buildLogic

melix commented:
I think this belongs to https://github.com/oracle/graalvm-reachability-metadata so that all projects benefit from it.

sdelamo commented:
> I think this belongs to https://github.com/oracle/graalvm-reachability-metadata so that all projects benefit from it.

I think it belongs to http://github.com/modelcontextprotocol/java-sdk but lets merge it here first so that Micronaut MCP works with native image.

melix commented:
Well, the point of the metadata repository is to add metadata to existing libraries, both us and  Spring folks contribute much here. While ideally each library should support GraalVM, I can understand that some wouldn't want to.

graemerocher commented on micronaut-mcp-server-java-sdk/src/main/resources/META-INF/native-image/io.micronaut.mcp.server.java.sdk/reflect-config.json:
```diff
@@ -0,0 +1,70 @@
+[
+  {"name": "ch.qos.logback.core.ConsoleAppender", "allDeclaredConstructors": true, "methods": [ { "name": "setTarget", "parameterTypes": ["java.lang.String"] } ]},
```
this logback config shouldn't be here

sdelamo replied:
should I move it micronaut-logging?

sdelamo replied:
Or in the GraalVM reachability metadata? I would prefer to have it here temporarily to ensure we have a native-image working version of Micronaut MCP with tests until it is in a published version of the GraalVM reachability metadata and we can remove the reflection configuration from this repository.

graemerocher replied:
it can't be here and it wrong that it is because this module as nothing to do with logback

graemerocher replied:
it is already in the reachability metadata for logback https://github.com/oracle/graalvm-reachability-metadata/blob/124d15d2cedec4f3f0033d8a0916890deb0d5c8f/metadata/ch.qos.logback/logback-classic/1.4.1/reflect-config.json#L365

Which probably means the tests are not using reachability metadata?

sdelamo replied:
> Which probably means the tests are not using reachability metadata?

They use Micronaut Application Gradle plugin which uses the GraalVM reachability Metadata repository by default as far as I know. 

I have removed `<target>System.err</target>` as these tests use HTTP transport and thus we can log to STDOUT. This allows us to remove the necessary GraalVM reflection configuration to invoke `setTarget`. I will send a PR to the GraalVM reachability metadata.

sdelamo committed:
remove <target>System.err</target>

Remove `<target>System.err</target>` as these tests use HTTP transport and thus we can log to STDOUT. This allows us to remove the necessary GraalVM reflection configuration to invoke setTarget.
